libs/core/langchain_core/tracers/base.py

```diff
@@ -37,13 +37,28 @@ class BaseTracer(_TracerCore, BaseCallbackHandler, ABC):
     def _persist_run(self, run: Run) -> None:
         """Persist a run."""
 
+    def _store_tool_call_metadata(self, run: Run) -> None:
```

Comment:
I'd put this all in `_complete_llm_run` in tracers/core. We don't need to do this on all runs IMO